lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb

```diff
@@ -0,0 +1,139 @@
+require 'fileutils'
+
+Puppet::Type.type(:bootstrap_rpm).provide(:bootstrap_rpm) do
+
+  commands :rpmbuild => 'rpmbuild'
+  commands :rpm => 'rpm'
+
+  def create
+    copy_sources
+    write_specfile
+    build_rpm
+    copy_rpm
+    link_rpm
+  ensure
+    FileUtils.remove_dir(base_dir) if File.directory?(base_dir)
+    @base_dir = nil
+  end
+
+  def delete
+  end
+
+  def exists?
+    !resource[:symlink] || File.exist?(resource[:symlink])
+  end
+
+  private
+
+  def base_dir
+    @base_dir ||= Dir.mktmpdir
+  end
+
+  def sources_dir
+    mkdir('SOURCES')
+  end
+
+  def spec_dir
+    mkdir('SPECS')
+  end
+
+  def rpm_dir
+    mkdir('RPMS')
+  end
+
+  def copy_sources
+    FileUtils.copy(resource[:script], File.join(sources_dir, File.basename(resource[:script])))
+  end
+
+  def write_specfile
+    File.open(File.join(spec_dir, "#{resource[:name]}.spec"), 'w') do |file|
+      file.write(spec)
+    end
+  end
+
+  def release
+    if !latest_rpm.nil? && File.exist?(latest_rpm)
+      rel = rpm('-qp', latest_rpm, "--queryformat=%{release}")
+      rel.to_i + 1
+    else
+      1
+    end
+  end
+
+  def build_rpm
+    output = rpmbuild(
+      '-ba',
+      File.join(spec_dir, "#{resource[:name]}.spec"),
+      '--define', "_topdir #{base_dir}",
+      '--define', "name #{resource[:name]}",
+      '--define', "release #{release}"
+    )
+
+    Puppet.debug(output)
```

Comment:
I wonder if Puppet can't do this by default when it calls commands. Perhaps if you use `--trace`?

Reply:
I don't know. How does that work through the installer? This felt like a nice easy option.

Reply:
Not sure. It just feels that this is hard to trace down. Just some debug output without knowing where it comes from.